src/phoebus_utils/programming_utils.hpp

```diff
@@ -0,0 +1,49 @@
+// © 2022. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+
+#ifndef PHOEBUS_UTILS_PROGRAMMING_UTILS_HPP_
+#define PHOEBUS_UTILS_PROGRAMMING_UTILS_HPP_
+
+#include <algorithm>
+#include <vector>
+
+namespace programming {
+
+template <class SPECIAL, template <class...> class TEMPL>
+struct is_specialization_of : public std::false_type {};
+
+template <template <class...> class TEMPL, class... TPARAMS>
+struct is_specialization_of<TEMPL<TPARAMS...>, TEMPL> : public std::true_type {};
```

Comment:
I understand what this is for and why it has utility. But... YUCK.

Reply:
It actually works really well. But I have no idea what this code is doing.

Reply:
It is taken almost directly from this standard lib proposal: http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2020/p2098r0.pdf